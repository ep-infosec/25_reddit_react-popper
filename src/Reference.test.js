// @flow
import React from "react";
import { mount } from "enzyme";

// Public API
import { Reference } from ".";

// Private API
import { ManagerContext } from "./Manager";

describe("Arrow component", () => {
  it("renders the expected markup", () => {
    const getReferenceRef = jest.fn();

    // HACK: wrapping DIV needed to make Enzyme happy for now
    const wrapper = mount(
      <div>
        <ManagerContext.Provider
          value={{
            getReferenceRef,
            referenceNode: undefined,
          }}
        >
          <Reference>{({ ref }) => <div ref={ref} />}</Reference>
        </ManagerContext.Provider>
      </div>
    );
    expect(wrapper.children()).toMatchSnapshot();
  });

  it("consumes the ManagerContext from Manager", () => {
    const getReferenceRef = jest.fn();

    // HACK: wrapping DIV needed to make Enzyme happy for now
    mount(
      <div>
        <ManagerContext.Provider
          value={{
            getReferenceRef,
            referenceNode: undefined,
          }}
        >
          <Reference>{({ ref }) => <div ref={ref} />}</Reference>
        </ManagerContext.Provider>
      </div>
    );
    expect(getReferenceRef).toHaveBeenCalled();
  });
});
