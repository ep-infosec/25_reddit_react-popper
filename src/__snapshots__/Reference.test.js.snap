// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow component renders the expected markup 1`] = `
<Provider
  value={
    Object {
      "getReferenceRef": [MockFunction] {
        "calls": Array [
          Array [
            <div />,
          ],
        ],
      },
      "referenceNode": undefined,
    }
  }
>
  <Reference>
    <Consumer>
      <InnerReference
        getReferenceRef={
          [MockFunction] {
            "calls": Array [
              Array [
                <div />,
              ],
            ],
          }
        }
      >
        <div />
      </InnerReference>
    </Consumer>
  </Reference>
</Provider>
`;
